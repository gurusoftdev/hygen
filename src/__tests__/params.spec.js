const params = require('../params')
const ftest = require('../test/ftest')
describe('params', () => {
  beforeEach(() => {
    process.env.HYGEN_TMPLS = null
  })
  ftest(
    'underscored templates',
    {
      app: {
        _templates: {
          some: 'file'
        }
      }
    },
    () => {
      const args = params('app', 'dont-take-this', ['foo', 'bar', 'baz'])
      expect(args).toMatchSnapshot()
    }
  )

  ftest(
    'env var overrides local templates',
    {
      app: {
        _templates: {
          some: 'file'
        },
        tmpls: {
          some: 'otherfile'
        }
      }
    },
    () => {
      process.env.HYGEN_TMPLS = 'tmpls'
      const args = params('app', 'dont-take-this', ['foo', 'bar', 'baz'])
      expect(args).toMatchSnapshot()
    }
  )

  ftest(
    'takes default folder',
    {
      app: {
        tmpls: {
          some: 'otherfile'
        },
        famboozle: {
          some: 'stupidfile'
        }
      }
    },
    () => {
      const args = params('app', 'famboozle', ['foo', 'bar', 'baz'])
      expect(args).toMatchSnapshot()
    }
  )
})
