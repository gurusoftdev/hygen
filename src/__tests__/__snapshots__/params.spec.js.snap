// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`params env var overrides local templates 1`] = `
Object {
  "action": undefined,
  "generator": "dont-take-this",
  "templates": "app",
}
`;

exports[`params takes default folder 1`] = `
Object {
  "action": undefined,
  "generator": "famboozle",
  "templates": "app",
}
`;

exports[`params underscored templates 1`] = `
Object {
  "action": undefined,
  "generator": "dont-take-this",
  "templates": "app",
}
`;
